### ISSUE TITLE
Document: Instruction pipeline and byte arrays

### ISSUE THREAD

**Issue by mankinskin** [solana-labs/solana#16034]
#### Problem

The [documentation](https://docs.solana.com/developing/programming-model/transactions#instruction-format) mentions an arbitrary byte array in instructions. I could not find any info about the size limit of these byte arrays or how they were treated by validators, so I opened [this thread](https://forums.solana.com/t/how-large-can-transactions-be/1247) on the forums.

#### Proposed Solution

Document the byte arrays in more detail, how big they may be, if/how this may change in the future, if these byte arrays are recorded into blocks and when they get passed to programs.

Basically this whole pipeline seems very important for developers because they will use these instructions to send messages between programs, so this should be well documented I believe.
